src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    const indexOfTarget = Animal.alive.indexOf(target);
+
+    if (target.hidden === false) {
+      target.health -= 50;
+    }
+
+    if (target.health === 0) {
+      Animal.alive.splice(indexOfTarget, 1);
```

Comment:
Try to avoid nesting loops (indexOf + splice), use filter instead

Reply:
if this animal already died, you anyway delete the last animal from the array. indexOf return -1 if the animal already died, and you will splice the last animal - splice(-1, 1)

you have this point in the checklist )